### ISSUE TITLE
Update mainline navigation and remove unused components and files after "landing pages" are merged

### ISSUE THREAD

**Issue by marcellamaki** [learningequality/kolibri#12729]
## Overview 
For our information architecture restructuring, we are going to be "squishing together" some of our current pages and components to move us to a different page naming system and organization. 

### Prerequisites
- Run Kolibri or `kolibri-demo` locally to get familiar with the current state of the UX/UI. Make sure you are rebased on top of develop and that you see the new `lessonstemp` path.

Before this can be worked on, the following PRs must be closed: 

- [ ] https://github.com/learningequality/kolibri/pull/12704
- [ ] https://github.com/learningequality/kolibri/pull/12730

## Acceptance Criteria

The goals of this PR are to broadly update the coach navigation and routes to be inline with the updated information architecture spec, and to clean up unused and duplicate code. There will likely be some amount of UI work happening in parallel or in follow up, for example open issues regarding filtering by groups (#12706) and csv export (#12474). Small UI bugs and updates of this nature are out of scope. 

- [ ]  Coach pages names and routes have been updated: Lessons, Quizzes, Learner, and Groups are now available on the main navigation - top nav, side nav, app nav.  _(Note: please discuss the plan for `lessonstemp` with Marcella before implementing. It may be helpful to maintain the two lesson paths available as we continue to work on the MVP lesson resource selection work)_
- [ ] The new navigation and pages align with [the "MVP" spec for each page on Figma](https://www.figma.com/design/lVJt5ukOrxS9qay0rXy7GC/0.18-Coach-updates?node-id=94-47233&node-type=canvas&t=97dlEXLQLkWnL9L8-0)
- [ ] Unused or duplicate components are deleted, and remaining/new components are renamed as needed to provide more clarity based on the new information architecture. Please suggest a list of "remapping"/renaming components for confirmation/feedback before finalizing. 
- [ ] **When the PR is ready, please share the built asset with @jtamiace (ask if she would prefer pex, dmg, etc.) for design QA and approval**, in addition to QA team review
